pkg/diagnostics/networkpod/pod.go

```diff
@@ -0,0 +1,206 @@
+		if (vmap == nil) || (vmap != nil && vmap[pod.Namespace] == api.GlobalVNID) {
+			globalPods = append(globalPods, pod)
+		} else {
+			nonGlobalPods = append(nonGlobalPods, pod)
+		}
+	}
+	return globalPods, nonGlobalPods
+}
+func checkDifferentNodePodToPodConnection(localGlobalPods, localNonGlobalPods, nonlocalGlobalPods, nonlocalNonGlobalPods []kapi.Pod, vmap map[string]uint32, r types.DiagnosticResult) {
+	// Applicable to flat and multitenant networks
+	if len(localGlobalPods) > 0 && len(nonlocalGlobalPods) > 0 {
+		checkPodToPodConnection(localGlobalPods[0], nonlocalGlobalPods[0], vmap, r)
+	} else {
+		r.Warn("DPodNet1005", nil, fmt.Sprintf("Skipping pod connectivity test for global projects on different nodes. Reason: Couldn't find 2 global pods."))
+	}
+
+	// Applicable to multitenant network
+	isMultitenant := (vmap != nil)
+	if isMultitenant {
+		if len(localNonGlobalPods) > 0 && len(nonlocalNonGlobalPods) > 0 {
+			checkPodToPodConnection(localNonGlobalPods[0], nonlocalNonGlobalPods[0], vmap, r)
+		} else {
+			r.Warn("DPodNet1006", nil, fmt.Sprintf("Skipping pod connectivity test for non-global projects on different nodes. Reason: Couldn't find 2 non-global pods."))
+		}
+
+		if len(localGlobalPods) > 0 && len(nonlocalNonGlobalPods) > 0 {
+			checkPodToPodConnection(localGlobalPods[0], nonlocalNonGlobalPods[0], vmap, r)
+		} else {
+			r.Warn("DPodNet1007", nil, fmt.Sprintf("Skipping pod connectivity test between global to non-global projects on different nodes. Reason: Couldn't find global and/or non-global pod."))
+		}
+	}
+}
+
+func checkSameNodePodToPodConnection(globalPods, nonGlobalPods []kapi.Pod, vmap map[string]uint32, r types.DiagnosticResult) {
+	// Applicable to both flat and multitenant networks
+	if len(globalPods) >= 2 {
+		checkPodToPodConnection(globalPods[0], globalPods[1], vmap, r)
+	} else {
+		r.Warn("DPodNet1008", nil, fmt.Sprintf("Skipping pod connectivity test for global projects on the same node. Reason: Couldn't find 2 global pods."))
+	}
+
+	// Applicable to multitenant network
+	isMultitenant := (vmap != nil)
+	if isMultitenant {
+		if len(nonGlobalPods) >= 2 {
+			checkPodToPodConnection(nonGlobalPods[0], nonGlobalPods[1], vmap, r)
+		} else {
+			r.Warn("DPodNet1009", nil, fmt.Sprintf("Skipping pod connectivity test for non-global projects on the same node. Reason: Couldn't find 2 non-global pods."))
+		}
+
+		if len(globalPods) > 0 && len(nonGlobalPods) > 0 {
+			checkPodToPodConnection(globalPods[0], nonGlobalPods[0], vmap, r)
+		} else {
+			r.Warn("DPodNet1010", nil, fmt.Sprintf("Skipping pod connectivity test between global to non-global projects on the same node. Reason: Couldn't find global and/or non-global pod."))
+		}
+	}
+}
+
+// Determine expected connection status for the given pods
+// true indicates success and false means failure
+func expConnStatus(ns1, ns2 string, vmap map[string]uint32) bool {
+	// Check if sdn is flat network
+	if vmap == nil {
+		return true
+	} // else multitenant
```

Comment:
(again, if we know that api.GlobalVNID is 0, then you don't need this check, because the following check will always be true when vmap is nil)